lib/transport/selenium-webdriver/method-mappings.js

```diff
@@ -330,8 +330,17 @@ module.exports = class MethodMappings {
         // Frames
         ///////////////////////////////////////////////////////////
         async switchToFrame(frameId) {
-          if (frameId === undefined) {
-            frameId = null;
+          try {
+            if (frameId === undefined) {
+              frameId = null;
+            } else if (isString(frameId)) {
+              frameId = await this.driver.findElement(By.css(`#${frameId}, *[name="${frameId}"]`));
```

Comment:
it's important that we remove this request from this method on move it somewhere in the command. As much as it's possible, we need to keep the methods here as simple mappings to requests in selenium, without much logic.

Reply:
@harshit-bs my team and I are looking forward to your fix. Are you able to make the suggested changes by @beatfactor please?

Reply:
Hey @literallyMello I have pushed the changes #[53921e0](https://github.com/nightwatchjs/nightwatch/pull/3135/commits/53921e06cfa9de0d982a3e8984de591927ac193d), a bit late due to health concerns. But sorry for inconvenience.

Reply:
@harshit-bs thanks for all that you do! We appreciate you getting this fixed. Thank you.